src/components/table/schedule/use-schedule-state.ts

```diff
@@ -10,34 +10,41 @@ import { MetadataLogic } from "../../../logic/schedule-logic/metadata.logic";
 
 // eslint-disable-next-line @typescript-eslint/class-name-casing
 export interface useScheduleStateReturn {
+  scheduleLogic: ScheduleLogic;
   scheduleLocalState: ScheduleComponentState;
   setNewSchedule: (scheduleModel: ScheduleDataModel) => void;
 }
 
 export const ScheduleLogicContext = React.createContext<ScheduleLogic | null>(null);
 
-export function useScheduleState(): useScheduleStateReturn {
+export function useScheduleState(
+  initialScheduleModelState: ScheduleDataModel
+): useScheduleStateReturn {
   const dispatchGlobalState = useDispatch();
   const [scheduleLocalState, setScheduleLocalState] = useState<ScheduleComponentState>(
     scheduleInitialState
   );
+  const scheduleLogic = useState<ScheduleLogic>(
```

Comment:
Wyniosłem logiką tutaj, ponieważ tworzenie nowego obiektu dla każdej zmiany było złym pomysłem:
logika była wsadzana do kontekstu i z niej potem korzystało dość dużo różnych komponentów. 
Przy zmianie referenceji do logiki, te części renderowały się od nowa i nie było możliwości to konrtolować. Dlatego zrobiłem tak, żeby referenja się nie zmieniała :D